src/java/net/jpountz/lz4/LZ4BlockInputStream.java

```diff
@@ -68,6 +68,11 @@ public LZ4BlockInputStream(InputStream in, LZ4FastDecompressor decompressor, Che
     this.compressedBuffer = new byte[HEADER_LENGTH];
     o = originalLen = 0;
     finished = false;
+    stopOnEmptyBlock = true;
+  }
+
+  protected void setStopOnEmptyBlock(boolean stopOnEmptyBlock) {
```

Comment:
I reconsider this and I think how about setting this in the constructor? I couldn't imagine users change this flag during the decompression.